src/client/partition.rs

```diff
@@ -115,6 +115,17 @@ impl PartitionClient {
         self.partition
     }
 
+    /// Re-run the leader discovery process for this client, and establish a
+    /// connection to the leader.
+    pub(crate) async fn refresh_leader(&self) -> Result<()> {
+        // Remove the current broker connection, if any.
+        *self.current_broker.lock().await = None;
+        // And acquire a new one, forcing discovery and establishing a cached
+        // connection to the leader.
+        self.get().await?;
+        Ok(())
+    }
```

Comment:
Since this is only used in combination w/ `PartitionClient::new` I would inline this function into `new`. This also spares you the `*self.current_broker.lock().await = None;` bit.